cmd/juju/status/status_internal_test.go

```diff
@@ -3717,11 +3717,33 @@ func (sm startMachineWithHardware) step(c *gc.C, ctx *context) {
 	cfg, err := ctx.st.ControllerConfig()
 	c.Assert(err, jc.ErrorIsNil)
 	inst, _ := testing.AssertStartInstanceWithConstraints(c, ctx.env, environscontext.NewCloudCallContext(), cfg.ControllerUUID(), m.Id(), cons)
-	err = m.SetProvisioned(inst.Id(), "fake_nonce", &sm.hc)
+	err = m.SetProvisioned(inst.Id(), "", "fake_nonce", &sm.hc)
 	c.Assert(err, jc.ErrorIsNil)
 	ctx.pingers[m.Id()] = pinger
 }
 
+type startAliveMachineWithDisplayName struct {
```

Comment:
You don't need all the code duplication - you can just put the displayname attribute on startAliveMachine and embed that struct in this one.